File: app/models/swars/style_info.rb
module Swars
  class StyleInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: "王道",   segment: :majority, },
      { key: "準王道", segment: :majority, },
      { key: "準変態", segment: :minority, },
      { key: "変態",   segment: :minority, },
    ]
  end
end

File: nuxt_side/components/Swars/models/style_info.js
import { ApplicationMemoryRecord } from "@/components/models/application_memory_record.js"

export class StyleInfo extends ApplicationMemoryRecord {
  static get define() {
    return [
      { key: "王道",   },
      { key: "準王道", },
      { key: "準変態", },
      { key: "変態",   },
    ]
  }
}
